### PR TITLE
edits to time units docs

### LINKED ISSUE #284: read new section on time units
You can read the new section I wrote at
  https://tskit.dev/pyslim/docs/latest/time_units.html

The last section ("Converting to SLiM time and back") is not new (although I made edits); however, the stuff before that is new and should have a wide-ish audience. It'd be nice if someone had a read of it to see if you agree and if it makes sense. @bhaller?

### PR DESCRIPTION
**Pull request by petrelharp**
This might close #284 (depends if others have edits there).

### CODE REVIEW

File: docs/time_units.md [hunk 1]
```diff
@@ -254,7 +265,7 @@ Mostly, you don't have to convert between the two,
 unless you want to match up information in a tree sequence
 with information written out by SLiM itself.
 In other words, SLiM's time counter measures the number of time steps
-("tick") since the start of the simulation,
+("ticks") since the start of the simulation,
```
Comment (andrewkern) [inline docs/time_units.md:268]: wondering about the simulation above? would it be better to use the harmonic mean of those gen times?

File: docs/time_units.md [hunk 2]
```diff
@@ -84,7 +91,7 @@ since one unit of time in the tree sequence equals 1/30th of a generation,
 this is one coalescence per pair of lineages every 600,000 generations,
```
Comment (andrewkern) [inline docs/time_units.md:91]: should this be every 600,000 _ticks_?

File: docs/time_units.md [hunk 3]
```diff
@@ -266,9 +277,9 @@ without needing to actually do any conversions yourself.
 SLiM's time counter is called the "tick".
```
Comment (andrewkern) [inline docs/time_units.md:277]:
```suggestion
As we keep mentioning, SLiM's time unit is called a"tick".
```

File: docs/time_units.md [hunk 4]
```diff
@@ -266,9 +277,9 @@ without needing to actually do any conversions yourself.
...
 However, the order of the life cycle differs between WF and nonWF models:
-in a WF model, it is "*first* {math}`\to` "*early* {math}`\to` *birth* {math}`\to` *late*",
-while in a nonWF model, it is "*first* {math}`\to` "*birth* {math}`\to` *early* {math}`\to` *late*".
-Usually, the first set of individuals are created in the *early()* phase of tick 1,
+in a WF model, it is *first* {math}`\to` *early* {math}`\to` *birth* {math}`\to` *late*",
+while in a nonWF model, it is *first* {math}`\to` *birth* {math}`\to` *early* {math}`\to` *late*.
+Often, the first set of individuals are created in the *early()* phase of tick 1,
 and so in a WF model reproduce immediately, in the same tick they were "born".
 Parents and offspring cannot have the same birth time in the tree sequence,
 and so some clever bookkeeping was required.
```
Comment (andrewkern) [inline docs/time_units.md:285]:
```suggestion
and so careful bookkeeping is required.
```

File: docs/time_units.md [hunk 5]
```diff
@@ -281,16 +292,38 @@ SLiM also keeps track of "how many birth phases of the life cycle have happened
...
 The extra wrinkle this introduces is that the correspondence between "tskit time ago"
 and "SLiM time" depends on *which phase the tree sequence was recorded in*.
 
+When the tree sequence is written out, SLiM records the value of its current tick
+and the current stage,
+which can be found in the metadata: ``ts.metadata['SLiM']['tick']``,
+and ``ts.metadata['SLiM']['stage']``.
+The "SLiM time" referred to by a ``time`` in the tree sequence
+(i.e., the value that would be reported by ``community.tick``
+within SLiM at the point in time thus referenced)
```
Comment (andrewkern) [inline docs/time_units.md:311]:
```suggestion
within SLiM at the point specified)
```

File: docs/time_units.md [hunk 6]
```diff
@@ -281,16 +292,38 @@ SLiM also keeps track of "how many birth phases of the life cycle have happened
...
+can be obtained by subtracting ``time`` from ``ts.metadata['SLiM']['tick']``
+and possibly adding or subtracting one depending on the current stage.
+The fiddly details are worked out by the {func}`.slim_time` method,
```
Comment (andrewkern) [inline docs/time_units.md:314]:
```suggestion
*Importantly*, from a users perspective all the difficult details are worked out by the {func}`.slim_time` method,
```

File: docs/time_units.md [hunk 7]
```diff
@@ -281,16 +292,38 @@ SLiM also keeps track of "how many birth phases of the life cycle have happened
...
+which takes a ``stage`` argument.
+
+Some of the other methods in pyslim -- those that depend on {func}`.individuals_alive_at`
+-- need you to tell them during which stage the tree sequence was saved with ``sim.treeSeqOutput``,
+and need this to be the same as the stage that any individuals were saved with ``sim.treeSeqRememberIndividuals``.
+(This can add another factor of plus or minus one to the calculations.)
+This argument, ``remembered_stage``, defaults to *late()*;
+we recommend that you also default to always Remembering individuals, and saving out the tree sequence,
+during *late()* as well, unless you have good reason not to.
+
```
Comment (andrewkern) [inline docs/time_units.md:324]: what follows below is technical enough that you should probably add it as reference subsection (clickable on the RHS of the page) and maybe call it something like "Details of SLiM time conversion" or something similar

**Review by andrewkern — COMMENTED** [review]
some comments here but looks pretty good. i also have some edits i can put on a separate PR

**Comment by petrelharp** [comment]
I rolled #295 into here also, thanks @andrewkern!

**Pull request merged**